lib/adiwg/mdtranslator/readers/iso19115_2/modules/module_time_period.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+require 'nokogiri'
+require 'adiwg/mdtranslator/internal/internal_metadata_obj'
+
+module ADIWG
+   module Mdtranslator
+      module Readers
+         module Iso191152
+            module TimePeriod
+               @@idAttr = 'gml:id'
+               @@beginPosXPath = 'gml:beginPosition | gml:begin'
+               @@endPosXPath = 'gml:endPosition | gml:end'
+               def self.unpack(xTimePeriod, hResponseObj)
+                  intMetadataClass = InternalMetadata.new
+                  hTimePeriod = intMetadataClass.newTimePeriod
+
+                  timeId = xTimePeriod.attr(@@idAttr)
+                  if timeId.nil?
+                     msg = 'ERROR: ISO19115-2 reader: Attribut gml:id is missing in gml:TimePeriod'
+                     hResponseObj[:readerExecutionMessages] << msg
+                  end
+
+                  hTimePeriod[:timeId] = timeId
+                  
+                  # :beginPosition/:begin and :endPosition/:end
+                  # <sequence>
+                  #    <choice>
+                  #      <element name="beginPosition" type="gml:TimePositionType"/>
+                  #      <element name="begin" type="gml:TimeInstantPropertyType"/>
+                  #    </choice>
+                  #    <choice>
+                  #      <element name="endPosition" type="gml:TimePositionType"/>
+                  #      <element name="end" type="gml:TimeInstantPropertyType"/>
+                  #    </choice>
+                  #    <group ref="gml:timeLength" minOccurs="0"/>
+                  # </sequence>
+                  xStartDatetime = xTimePeriod.xpath(@@beginPosXPath)[0]
+                  startDatetime = intMetadataClass.newDateTime
+                  if !xStartDatetime.nil? && !xStartDatetime.text.strip.empty?
+                     dt = AdiwgDateTimeFun.dateTimeFromString(xStartDatetime.text)
+                     startDatetime[:dateTime] = dt[0].to_s
+                     startDatetime[:dateResolution] = dt[1]
+                  else
+                     msg = 'ERROR: ISO19115-2 reader: Element gml:beginPosition or gml:begin '\
+                           'is missing in gml:TimePeriod'
+                     hResponseObj[:readerExecutionMessages] << msg
```

Comment:
do we need to add `hResponseObj[:readerExecutionPass] = false` here if it's required?

Reply:
i've been doing that. frankly, i'd like to go back and put a little more thought into the reader execution, validation, and structure messages ( e.g. am I storing the warning/error in an array which makes sense? ) and the logs themselves ( e.g. when is it an error vs warning? )

Reply:
@Jin-Sun-tts please also add `hResponseObj[:readerExecutionPass] = false` under the array append